feat(service): render static html in spa build

devCrossNet opened the pull request:
closes #26

<!--
There are two main goals in this document, depending on the nature of your PR:

- description: please tell us about your PR
- checklist: please review the checklist

To help to quickly understand the nature of your pull request,
please create a description that incorporates the following elements:
-->

### What is accomplished by your PR?

This PR will add a new flag to the build command `--static` which is the same as `--spa`, the spa build will be replaced at some point with the static build.

The static build allows our users to define an array of routes that should be rendered as part of the build. instead of serving an empty index.html we will now server the content of the `/` route by default. it's already possible to render multiple static pages. they can be defined in the .vuesion/config under the key `spa.routesToRender` e.g. `["/", "/form"]`

### Is there something controversial in your PR?
I am not quite sure if the key `spa.routesToRender` is good enough, I am open for suggestions.

### Link to the Issue
#26 

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [ ] Add new passing unit tests to cover the code introduced by your PR

<!--
Thanks for contributing!
-->

MohitTilwani15 commented:
@devCrossNet  I think the looping of routes (`for (const route of routes)`) should be in a try catch to show errors with meaningful text like when user enters a route in config which doesn't exits.

devCrossNet commented:
@MohitTilwani15 I changed it to more detailed error messages.
<img width="704" alt="Bildschirmfoto 2019-04-19 um 16 39 16" src="https://user-images.githubusercontent.com/1667598/56432039-48752d00-62cc-11e9-83a5-34e37ea3b889.png">


 I also removed the static flag. I think to put static rendering under single-page-application is totally fine.

I also changed the config slightly to give users the possibility to decide which route should be rendered as index.html
